test/periphery/CardPaymentProcessorUpgradeable.test.ts

```diff
@@ -0,0 +1,1421 @@
+        []
+      )).to.be.revertedWith(REVERT_MESSAGE_IF_INPUT_ARRAY_OF_AUTHORIZATION_IDS_IS_EMPTY);
+    });
+
+    it("Is reverted if one of the payment authorization IDs is zero", async () => {
+      await expect(cardPaymentProcessor.connect(admin).unclearPayments(
+        [authorizationIds[0], ZERO_BYTES16_STRING]
+      )).to.be.revertedWith(REVERT_MESSAGE_IF_PAYMENT_AUTHORIZATION_ID_IS_ZERO);
+    });
+
+    it("Is reverted if one of the payments with provided authorization IDs does not exist", async () => {
+      await expect(cardPaymentProcessor.connect(admin).unclearPayments(
+        [
+          authorizationIds[0],
+          createBytesString(payments[payments.length - 1].authorizationId + 1, BYTES16_LENGTH)
+        ]
+      )).to.be.revertedWith(REVERT_MESSAGE_IF_PAYMENT_DOES_NOT_EXIST);
+    });
+
+    it("Is reverted if one of the payments is uncleared", async () => {
+      await proveTx(cardPaymentProcessor.connect(admin).unclearPayment(
+        authorizationIds[0]
+      ));
+      await expect(cardPaymentProcessor.connect(admin).unclearPayments(
+        authorizationIds
+      )).to.be.revertedWith(REVERT_MESSAGE_IF_PAYMENT_IS_UNCLEARED);
+    });
+
+    it("Does not transfer tokens, emits the correct events, changes the state properly", async () => {
+      await checkCardPaymentProcessorState(payments);
+      await expect(cardPaymentProcessor.connect(admin).unclearPayments(
+        authorizationIds
+      )).to.changeTokenBalances(
+        brlcMock,
+        [cardPaymentProcessor, ...accountAddresses],
+        [0, ...accountAddresses.map(() => 0)]
+      ).and.to.emit(
+        cardPaymentProcessor,
+        "UnclearPayment"
+      ).withArgs(
+        authorizationIds[0],
+        payments[0].account.address,
+        payments[0].amount,
+        expectedClearedBalances[0],
+        expectedUnclearedBalances[0],
+        payments[0].revocationCounter || 0
+      ).and.to.emit(
+        cardPaymentProcessor,
+        "UnclearPayment"
+      ).withArgs(
+        authorizationIds[1],
+        payments[1].account.address,
+        payments[1].amount,
+        expectedClearedBalances[1],
+        expectedUnclearedBalances[1],
+        payments[1].revocationCounter || 0
+      );
+      payments.forEach(payment => payment.status = PaymentStatus.Uncleared)
+      await checkCardPaymentProcessorState(payments);
+    });
+  });
+
+  describe("Function 'revokePayment()'", async () => {
+    const reversingPaymentCorrelationId: string = SOME_BYTES16_STRING;
+    const parentTxHash: string = SOME_BYTES32_STRING;
+
+    let payment: TestPayment;
+    let authorizationId: string;
+    let admin: SignerWithAddress;
+
+    beforeEach(async () => {
+      payment = {
+        authorizationId: 987,
+        account: user1,
+        amount: 876,
+        status: PaymentStatus.Nonexistent,
+        revocationCounter: 0,
+        makingPaymentCorrelationId: 765,
+        parentTxHash: parentTxHash
```

Comment:
Codacy has a fix for the issue: [Insert `,`](https://app.codacy.com/gh/cloudwalk/brlc-token/pullRequest?prid=9781596)
```suggestion
        parentTxHash: parentTxHash,
```